src/GeneratedPlaylists/GeneratedPlaylist.test.tsx

```diff
@@ -33,14 +36,49 @@ describe('GeneratedPlaylist', () => {
       }
     ]
   }
+  const playlist2 = {
+    id: 1,
+    name: 'military western ska',
+    isSaved: true,
+    tracks: [
+      {
+        "mbid": "e2bad905-75a4-499d-bdea-2e916d73ad76",
+        "artist": {
+          "name": "Test artist",
+          "artistUrl": "https://www.last.fm/music/Sublime"
+        },
+        "duration": "208",
+        "songName": "Test song",
+        "songUrl": "https://www.last.fm/music/Sublime/_/Smoke+Two+Joints"
+      }
+    ]
+  }
   const selectedGenre = 'military western ska'
   beforeEach(() =>
 		render(
-      <GeneratedPlaylist selectedGenre={selectedGenre} playlist={playlist} />
+      <MemoryRouter>
+        <GeneratedPlaylist selectedGenre={selectedGenre} playlist={playlist1} toggleSaved={mockToggleSaved}/>
+      </MemoryRouter>
 		)
   );
   test('should render a generated playlist', () => {
     expect(screen.getByText('Smoke Two')).toBeInTheDocument();
     expect(screen.getByText('Forward March')).toBeInTheDocument();
-	});
+  });
+  test('should default a generated playlist to be unsaved', () => {
+    expect(screen.getByAltText('Save playlist')).toBeInTheDocument()
+  })
+  test('should allow a user to save playlist', () => {
+    userEvent.click(screen.getByAltText('Save playlist'))
+    expect(mockToggleSaved).toHaveBeenCalledWith(playlist1)
+    screen.debug()
```

Comment:
Looks great! Let's remember to remove all instances of console.log() or screen.debug() before the final commit for presentation.